lib/ansible/module_utils/junos.py

```diff
@@ -40,7 +40,7 @@
     'password': dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), no_log=True),
     'ssh_keyfile': dict(fallback=(env_fallback, ['ANSIBLE_NET_SSH_KEYFILE']), type='path'),
     'timeout': dict(type='int', default=10),
-    'provider': dict(type='dict'),
+    'provider': dict(type='dict', no_log=True),
```

Comment:
Not sure if that's needed, it's only the password that he should be masked out, which I believe is done already.